contrib/native/client/src/clientlib/channel.hpp

```diff
@@ -60,11 +66,24 @@ class UserProperties;
 
     class ChannelContext{
         public:
-            ChannelContext(DrillUserProperties* props):m_properties(props){};
+            ChannelContext(DrillUserProperties* props):m_properties(props),m_hasCertHostnameVerificationErr(false){};
             virtual ~ChannelContext(){};
             const DrillUserProperties* getUserProperties() const { return m_properties;}
+
+            /// @brief Check the certiticate hostname verification status.
+            /// 
+            /// @return if the verification has failed.
+            const bool HasCertHostnameVerificationError() const {return m_hasCertHostnameVerificationErr;}
```

Comment:
I think we should put these inside SSLChannelContext since these are SSL related items